### DISCUSSION TITLE
Distribution package cleanup (PyPI)

### DISCUSSION THREAD

**Discussion by MartinThoma** [py-pdf/pypdf#718] [General]
I would like to keep the distributed package as small as possible. Also I would like to keep the stuff within the PyPDF2 package minimal. The time I can spend on PyPDF2 is fixed - if there are things that are not so important but require attention/maintenance, the more important topics might get lost.

For this reason, I would like to remove the following from the distributed package:

1. :black_square_button:  **Resources**: Biggest in size, only PDF files. Only used by the tests.
2. :black_square_button:  **Tests**: I don't see a use-case where a user would NOT want the Github version of PyPDF2 AND want to run the tests.
3. :heavy_check_mark:  **Sample_Code**: That should be in the docs (maybe even just linked?), but I don't see that anybody would install PyPDF2 via pip and run the sample code from the package.
4. :black_square_button:  **Scripts**: Propably most controversial. `pdfcat`, `2-up.py`, and `pdf-image-extractor.py` are in there. 

For (4), I suggest that I create another repository for `pdfcat` within the py-pdf Github organization + upload it on PyPI. For the other two scripts, I would create one repository called `PyPDF2-examples`.

Creating the `PyPDF2-examples` repository can be done now. I would keep the stuff in PyPDF2 as is (without accepting any PRs that modify those) and remove it with PyPDF 2.0.0.

What do you think?

**Comment by MasterOdin** [comment]
I think that for (1), (2), and (3), it's probably safe to just remove these from a 1.x release, and then re-add if anyone complains. In my experience, most packages don't include this stuff anyway.

**Comment by MartinThoma** [comment]
I've just created cpdf so that users of the `Scripts` can still use them - and hopefully now in a more convenient way: https://github.com/py-pdf/cpdf

**Reply by sekrause** [reply]
I would suggest using a different name for this utility because of a naming conflict: Another command-line utility for processing PDF files called `cpdf` already exists:

* https://community.coherentpdf.com/
* https://github.com/coherentgraphics

**Comment by MartinThoma** [comment]
Done via https://github.com/py-pdf/PyPDF2/pull/845
